### PR TITLE
(PUP-10145) Add :puppet service

### PR DESCRIPTION
**Pull request by joshcooper**
Add serialization error and protected methods for serializing/deserializing.

Add a :puppet service to get nodes/catalogs and put facts.

### CODE REVIEW

File: lib/puppet/http/service.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class Puppet::HTTP::Service
   attr_reader :url
 
-  SERVICE_NAMES = [:ca, :report].freeze
+  SERVICE_NAMES = [:ca, :puppet, :report].freeze
```
Comment (melissa) [inline lib/puppet/http/service.rb]: nit: should this change be in this commit?

Comment (joshcooper) [reply lib/puppet/http/service.rb]: Updated

Comment (melissa) [reply lib/puppet/http/service.rb]: It looks like you haven't pushed up the change yet

**Comment by joshcooper** [comment]
nah, I think it's good

**Comment by melissa** [comment]
I'll merge once travis & appveyor finish :)

**Pull request merged**